doc/dev/debugging.rst

```diff
@@ -403,4 +403,44 @@ The recompilation starts after the given checkpoint stage.
 
     replace_ir(f, "HLOLoweringPass", new_ir)
     f(2.0)
-    >> 8.0
\ No newline at end of file
+
+C Executable Generation
+=================
+
+Catalyst provides a way to generate a c executable that calls the shared object of the given
+qjit-decorated function.
```

Comment:
Just a small comment here, but with manually written docs (like `debugging.rst`) I would approach it as follows:

- This page should act as a reference guide, giving the reader a 'lay of the land'/map of the available debugging functionality, highlighting all useful functionality with minimal examples (where appropriate!)

- However, we should avoid fully documenting the functions here (e.g., talking about keyword arguments, having complex examples), and instead direct the user to the docstring where this should live instead.

The reason for this is that:

- someone might end up on the docstring _without_ going through this page first, so good to centralize the documentation there.
- the docstring is (much) closer to the code, and is more likely to be updated when someone changes functionality of the function